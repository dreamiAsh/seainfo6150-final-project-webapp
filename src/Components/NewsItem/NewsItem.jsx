import React from 'react'
import styles from "./NewsItem.module.css";
import { Link, BrowserRouter } from "react-router-dom";

const NewsItem = (props) => {
    return (
        <div>
            <BrowserRouter forceRefresh={true}><Link className={styles.box} to={`/news/${props.listItem.title}/${props.listItem.date}/${props.listItem.imageId}/${props.listItem.content}`}>
                <img className={styles.img} src={`https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%${props.listItem.imageId}.jpg&w=272&h=272&c=sc&poi=face&q=85`} alt=""/>
                
                <div className={styles.rightBox}>
                    <h6 className={styles.title}>{props.listItem.title}</h6>
                    <p className={styles.date}>{props.listItem.date}</p>
                </div>
                
            </Link></BrowserRouter>
        </div>
    )
}

export default NewsItem
