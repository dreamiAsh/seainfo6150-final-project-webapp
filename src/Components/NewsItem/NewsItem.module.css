.box{
    color: black;
    text-decoration: none;
    border-bottom: solid 1px rgb(228, 228, 228);
    width: 76%;
    height: 100px;
    display: block;
    overflow: hidden;
    margin: 14px 14%;
}
.leftBox{
    float: left;
}
.rightBox{
    float: left;
}
.title{
    font-size: 12px;
    margin-top: 10px;
    float: left;
}
.title:hover{
    color: orange;
}
.date{
    font-size: 13px;
    margin-top: 10px;
    color: grey;
}
.img{
    width: 80px;
    float: left;
    margin-right: 40px;
}
@media (min-width: 600px) {
    .title {
        font-size: 15px;
    }
}
@media (min-width: 800px) {
    .title {
        font-size: 16px;
    }
}
@media (min-width: 1100px) {
    .title {
        font-size: 18px;
    }
}