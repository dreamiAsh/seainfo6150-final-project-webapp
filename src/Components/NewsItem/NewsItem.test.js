import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewsItem from "./NewsItem.jsx";

it("renders the NewsItem component correctly", () => {
  const listItem = {
      title: "a",
      date: "b",
      imageId: "c",
      content: "d"
  }  
  const { container } = render(<NewsItem listItem={listItem}/>);
  expect(container).toMatchSnapshot();
});