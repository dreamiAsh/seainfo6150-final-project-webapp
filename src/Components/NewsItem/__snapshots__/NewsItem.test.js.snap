// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the NewsItem component correctly 1`] = `
<div>
  <div>
    <a
      class="box"
      href="/news/a/b/c/d"
    >
      <img
        alt=""
        class="img"
        src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%c.jpg&w=272&h=272&c=sc&poi=face&q=85"
      />
      <div
        class="rightBox"
      >
        <h6
          class="title"
        >
          a
        </h6>
        <p
          class="date"
        >
          b
        </p>
      </div>
    </a>
  </div>
</div>
`;
